File: app/src/main/java/me/tatarka/betterfragment/sample/MyViewModel.java
package me.tatarka.betterfragment.sample;

import android.util.Log;

import androidx.lifecycle.ViewModel;

public class MyViewModel extends ViewModel {
    public MyViewModel() {
        Log.d("VIEWMODEL", "newInstance " + this);
    }

    @Override
    protected void onCleared() {
        Log.d("VIEWMODEL", "cleared " + this);
    }
}

File: app/src/main/java/me/tatarka/betterfragment/sample/MyViewModel.kt
package me.tatarka.betterfragment.sample

import android.util.Log

import androidx.lifecycle.ViewModel

class MyViewModel : ViewModel() {
    init {
        Log.d("VIEWMODEL", "newInstance ${this}")
    }

    override fun onCleared() {
        Log.d("VIEWMODEL", "cleared ${this}")
    }
}
